### ISSUE TITLE
Soft exit

### ISSUE THREAD

**Issue by kurbaitis** [anycable/anycable#34]
Hey,

One more time thanks for creating this amazing product.

I have a question about exiting RPC server and maybe `anycable-go` server.

The problem is about my callback `def unsubscribe` for specific channels.
When I terminate the process of the RPC server (`./bin/anycable`), I get something like this:

```
^C/Users/kristis/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/grpc-1.8.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_server.rb:391:in `request_call': Interrupt
	from /Users/kristis/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/grpc-1.8.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_server.rb:391:in `loop_handle_server_calls'
	from /Users/kristis/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/grpc-1.8.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_server.rb:352:in `run'
	from /Users/kristis/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/anycable-0.5.0/lib/anycable/server.rb:17:in `start'
	from ./bin/anycable:8:in `<main>'
E1223 22:23:43.886264000 140737055724480 completion_queue.cc:510]      assertion failed: cqd->completed_head.next == (uintptr_t)&cqd->completed_head
Abort trap: 6
```

I'd like to have something like soft exit, where the unsubscribe callbacks could be called before actual process exits. So for each existing subscription for a particular server, there should be connection terminating hook.

It this feature present and I can't see it? Maybe you see this as a nice feature to have?

Thanks

**Comment by palkan** [comment]
Hi!

This feature is currently missing. But it's the one of the main goals for the next release.
There are two features, actually: graceful shutdown (for both Ruby RPC and `anycable-go`) and handling RPC server unavailability (that would make it possible to restart RPC without disconnecting existing clients).

**Comment by palkan** [comment]
Closed by #58
